src/views/virtualmachinesinstance/details/tabs/console/components/AccessConsoles/AccessConsoles.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+
+import { Select, SelectOption, SelectOptionObject, SelectVariant } from '@patternfly/react-core';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Consoles/AccessConsoles';
+
+import {
+  DESKTOP_VIEWER_CONSOLE_TYPE,
+  NONE_TYPE,
+  SERIAL_CONSOLE_TYPE,
+  VNC_CONSOLE_TYPE,
+} from '../utils/ConsoleConsts';
+
+import {
+  AccessConsolesProps,
+  getChildTypeName,
+  getConsoleForType,
+  isChildOfType,
+} from './utils/accessConsoles';
+
+import '@patternfly/react-styles/css/components/Consoles/AccessConsoles.css';
+
+export const AccessConsoles: React.FC<AccessConsolesProps> = ({
+  children,
+  textSelectConsoleType = 'Select console type',
+  textSerialConsole = 'Serial console',
+  textVncConsole = 'VNC console',
+  textDesktopViewerConsole = 'Desktop viewer',
+  preselectedType = null,
+}) => {
+  const typeMap = {
+    [SERIAL_CONSOLE_TYPE]: textSerialConsole,
+    [VNC_CONSOLE_TYPE]: textVncConsole,
+    [DESKTOP_VIEWER_CONSOLE_TYPE]: textDesktopViewerConsole,
+  };
+
+  const [type, setType] = React.useState(
+    preselectedType !== NONE_TYPE
+      ? ({ value: preselectedType, toString: () => typeMap[preselectedType] } as SelectOptionObject)
+      : null,
+  );
+
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const selectOptions = React.Children.toArray(children as React.ReactElement[]).reduce(
+    (acc: any[], child: any) => {
+      if (isChildOfType(child, VNC_CONSOLE_TYPE)) {
+        acc.push(
+          <SelectOption
+            key={VNC_CONSOLE_TYPE}
+            id={VNC_CONSOLE_TYPE}
+            value={
+              { value: VNC_CONSOLE_TYPE, toString: () => textVncConsole } as SelectOptionObject
+            }
+          />,
+        );
+      } else if (isChildOfType(child, SERIAL_CONSOLE_TYPE)) {
+        acc.push(
+          <SelectOption
+            key={SERIAL_CONSOLE_TYPE}
+            id={SERIAL_CONSOLE_TYPE}
+            value={
+              {
+                value: SERIAL_CONSOLE_TYPE,
+                toString: () => textSerialConsole,
+              } as SelectOptionObject
+            }
+          />,
+        );
+      } else if (isChildOfType(child, DESKTOP_VIEWER_CONSOLE_TYPE)) {
+        acc.push(
+          <SelectOption
+            key={DESKTOP_VIEWER_CONSOLE_TYPE}
+            id={DESKTOP_VIEWER_CONSOLE_TYPE}
+            value={
+              {
+                value: DESKTOP_VIEWER_CONSOLE_TYPE,
+                toString: () => textDesktopViewerConsole,
+              } as SelectOptionObject
+            }
+          />,
+        );
+      } else {
+        const typeText = getChildTypeName(child);
+        acc.push(
+          <SelectOption
+            key={typeText}
+            id={typeText}
+            value={{ value: typeText, toString: () => typeText } as SelectOptionObject}
+          />,
+        );
+      }
```

Comment:
Can this be done like this?
```js
        const typeText = getChildTypeName(child);
        acc.push(
          <SelectOption
            key={typeText}
            id={typeText}
            value={{ value: typeText, toString: () => typeMap[typeText] || typeText } as SelectOptionObject}
          />,
        );
```

Reply:
great suggestion

Reply:
why do we maintain a copy of patternflys component?
https://github.com/patternfly/patternfly-react/blob/main/packages/react-console/src/components/AccessConsoles/AccessConsoles.tsx

if we need to add new features should we add them in patternfly?
if we actually want to maintain our own version, we should do it in a more public place, maybe notify the patently guys we are doing it?

Reply:
If we will need a new feature we can simply add it to our own code base, pattternfly/console codebase isn't up-to-date with the rest of the console code and causes conflicts. 
If we will want the far future to do it in a public place we can think over the approach.